### PR TITLE
RYA-151 - Benchmarking Tools

### PR DESCRIPTION
**Pull request by kchilton2**

### CODE REVIEW

File: extras/indexing/src/main/java/mvm/rya/indexing/external/PrecomputedJoinStorageSupplier.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import com.google.common.base.Optional;
 import com.google.common.base.Supplier;
 
+import cern.colt.Arrays;
```
Comment (amihalik) [inline extras/indexing/src/main/java/mvm/rya/indexing/external/PrecomputedJoinStorageSupplier.java]: Is this correct?  Or do you mean `java.util.Arrays`

File: extras/rya.benchmark/pom.xml [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    <build>
+        <resources>
+            <resource>
+                <directory>src/main/xsd</directory>
+            </resource>
+        </resources>
+    
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (amihalik) [inline extras/rya.benchmark/pom.xml]: this should be define in the project's root pom

File: extras/rya.benchmark/pom.xml [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    <artifactId>rya.benchmark</artifactId>
+
+    <name>Apache Rya Benchmarks</name>
+
+    <dependencies>
+        <!-- JMH Benchmark Framework dependencies -->
+        <dependency>
+            <groupId>org.openjdk.jmh</groupId>
+            <artifactId>jmh-core</artifactId>
+            <version>${jmh.version}</version>
```
Comment (amihalik) [inline extras/rya.benchmark/pom.xml]: These versions should be defined in the root pom as properties and in the root pom's plug-in management section

File: extras/indexing/src/main/java/mvm/rya/indexing/external/PrecomputedJoinStorageSupplier.java (hunk 1 above)
Comment (kchilton2) [inline extras/indexing/src/main/java/mvm/rya/indexing/external/PrecomputedJoinStorageSupplier.java]: That's likely a mistake. Good catch.

File: extras/rya.benchmark/pom.xml (hunk 2 above)
Comment (kchilton2) [inline extras/rya.benchmark/pom.xml]: Which part specifically?

Comment (amihalik) [inline extras/rya.benchmark/pom.xml]: Sorry for my terse comment.  These should be defined in the parent POM:

1. Versions for dependencies should be defined as parameters (e.g. like you've done with jmh.version).  You should also do this for maven plug-ins (e.g. you haven't done this for maven-clean-plugin).

2. Dependencies (and plug-ins) and their versions should be defined in the DependenciesManagement (and pluginManagement) sections of the root pom.

3. The root pom should be the only pom to define the java versions.  Subprojects shouldn't be using the maven-compiler-plugin to define their own versions of java.

4. You have some nice plug-ins that should be moved to the root pom as well (e.g. license-maven-plugin).  I think @DLotts is working on this, so there's no need for you to modify that here.

Comment (kchilton2) [inline extras/rya.benchmark/pom.xml]: This POM file was largely generated by as a JMH archetype and I'm not sure what I can safely change without breaking how the JMH stuff works.

Comment (kchilton2) [inline extras/rya.benchmark/pom.xml]: Fixed what I know can be moved. It's in the new commit.

File: extras/indexing/src/main/java/mvm/rya/indexing/external/PrecomputedJoinStorageSupplier.java (hunk 1 above)
Comment (kchilton2) [inline extras/indexing/src/main/java/mvm/rya/indexing/external/PrecomputedJoinStorageSupplier.java]: Fixed, new commit.

**Pull request closed without merging**